src/assignthree/util/DfsOperatorUtils.java

```diff
@@ -0,0 +1,94 @@
+package assignthree.util;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.Map;
+
+/*
+ * This class will perform all the searching based option depth wise
+ */
+public class DfsOperatorUtils {
+
+    /*
+     * dfs util function for the searching the node subtree
+     * @param src source node for the search
+     * @param child hashmap for nodeid to list
+     * @param res list for adding all node collected
+     * @param visited boolean array for visited array
+     */
+    public static void dfsHelper(final int src, final Map<Integer, ArrayList<Integer>> child,
+                                 final ArrayList<Integer> res, final boolean[] visited) {
+        visited[src] = true;
```

Comment:
what if src is not present

Reply:
range is high for visited array so src will be present